libsolidity/interface/CompilerStack.cpp

```diff
@@ -444,20 +445,35 @@ map<string, unsigned> CompilerStack::sourceIndices() const
 	return indices;
 }
 
-Json::Value const& CompilerStack::interface(string const& _contractName) const
+Json::Value const& CompilerStack::contractABI(string const& _contractName) const
 {
-	return metadata(_contractName, DocumentationType::ABIInterface);
+	return contractABI(contract(_contractName));
 }
 
-Json::Value const& CompilerStack::metadata(string const& _contractName, DocumentationType _type) const
+Json::Value const& CompilerStack::contractABI(Contract const& _contract) const
 {
 	if (m_stackState < AnalysisSuccessful)
 		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("Parsing was not successful."));
 
-	return metadata(contract(_contractName), _type);
+	solAssert(_contract.contract, "");
+	std::unique_ptr<Json::Value const>* abi = &_contract.abi;
```

Comment:
If `abi` can never be `null_ptr`, this can be a reference of (not a pointer to) a unique pointer.